### ISSUE TITLE
Request Body Parameters unclear for setting up AWS S3 via API

### ISSUE THREAD

**Issue by Pinna3** [ONLYOFFICE/CommunityServer#383]
I'm trying to write an API POST request to connect/save my AWS S3 bucket to my self-hosted Only Office server. This is easy enough to do manually so I presume it can be done via an API call but I'm unclear how to submit my AWS IAM access key / access secret and indicate AWS S3. Using requests my python function will look something like this albeit with a smaller json object, hopefully only 3 items...

```PYTHON
def connect_s3(auth_token):
    data = {
        "url": "some text",
        "login": "some text",
        "password": "some text",
        "token": "some text",
        "isCorporate": True,
        "customerTitle": "some text",
        "providerKey": "some text",
        "accessKey": "some text",
        "secretAccessKey": "some text"
    }
    r = requests.post(f'{URL_ROOT}/files/thirdparty', json=data, headers={"Content-Type": "application/json", "Authorization": f"Bearer {auth_token}"})  # noqa: E501
    return json.loads(r.content)
```
What options do I need to pass in that json object to indicate my access key / access secret and AWS S3 provider to the endpoint `POST api/2.0/files/thirdparty`? The API docs don't specify a `providerKey` for S3 despite it being available in the GUI.

**Comment by Pinna3** [comment]
Oh thats actually exactly what I'm trying to do. I want to use S3 for backup file storage, not for normal documents... Sounds like I'm targeting the wrong endpoint then. Is there an API endpoint for configuring S3 for backup storage? I see how to do it manually but I need to do it programmatically with the API. I was able to save a backup file on the local filesystem like so...
```PYTHON
def start_backup(auth_token):
    data = {
        "storageType": "Local",
        "storageParams": [
            {
                "Key": 'filePath',
                "Value": '/var/www/onlyoffice/Data/Studio/00/00/01/backup'
            }
        ],
        "backupMail": 'true'
    }
    r = requests.post(f'{URL_ROOT}/portal/startbackup', json=data, headers={"Content-Type": "application/json", "Authorization": f"Bearer {auth_token}"})  # noqa: E501
    return json.loads(r.content)
```
But I'd like to be able to pass my aws credentials, specify S3, and have the backup file go directly to S3 (then restore directly from S3). Is there a separate endpoint to configure S3 or maybe special keys for the json object sent to the `POST api/2.0/portal/startbackup` endpoint?

**Comment by Carazyda** [comment]
You can send it in request, for example
```
api/2.0/portal/startbackup.json
    "StorageType": "5",
    "storageParams":  [{"key":"module", "value":"S3"},
     {"key":"bucket", "value":"qwe"},
     {"key":"region", "value":"qwe"},],
    "backupMail" : false
```
But first you need configure keys for S3 in portal settings /Management.aspx?type=9

**Comment by Pinna3** [comment]
Thank you! Do you have a link to those Key: Value labels? I couldn't find them anywhere in the API docs...

**Comment by Carazyda** [comment]
Yes, the documentation of this method is far from ideal. You can see the parameter set here https://github.com/ONLYOFFICE/CommunityServer/blob/master/web/studio/ASC.Web.Studio/web.consumers.config#L176-L185

**Comment by Carazyda** [comment]
Sorry, but this is not currently implemented. We will try to add this in one of the future versions.
